File: filesystem/src/test/java/com/nytimes/android/external/fs/UtilTest.java
package com.nytimes.android.external.fs;

import org.junit.Test;

import java.io.ByteArrayInputStream;
import java.io.File;
import java.io.IOException;

import okio.BufferedSource;
import okio.Okio;

import static com.google.common.base.Charsets.UTF_8;
import static org.assertj.core.api.Assertions.assertThat;
import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.verify;
import static org.mockito.Mockito.when;

public class UtilTest {

    private final Util util = new Util();

    @Test
    public void testSimplifyPath() {
        assertThat(util.simplifyPath("/a/b/c/d")).isEqualTo("/a/b/c/d");
        assertThat(util.simplifyPath("/a/../b/")).isEqualTo("/b");
        assertThat(util.simplifyPath("/a/./b/c/../d")).isEqualTo("/a/b/d");
        assertThat(util.simplifyPath("./a")).isEqualTo("/a");
        assertThat(util.simplifyPath(null)).isEqualTo("");
        assertThat(util.simplifyPath("")).isEqualTo("");
    }

    @Test
    public void createParentDirTest() throws IOException {
        File child = mock(File.class);
        File parent = mock(File.class);
        when(child.getCanonicalFile()).thenReturn(child);
        when(child.getParentFile()).thenReturn(parent);
        when(parent.isDirectory()).thenReturn(true);
        util.createParentDirs(child);
        verify(parent).mkdirs();
    }

    static BufferedSource source(String data) {
        return Okio.buffer(Okio.source(new ByteArrayInputStream(data.getBytes(UTF_8))));
    }
}

File: filesystem/src/test/java/com/nytimes/android/external/fs3/UtilTest.kt
package com.nytimes.android.external.fs3

import com.nhaarman.mockitokotlin2.doReturn
import com.nhaarman.mockitokotlin2.whenever
import org.assertj.core.api.Assertions.assertThat
import org.junit.Test
import org.mockito.Mockito.mock
import org.mockito.Mockito.verify
import java.io.File
import java.io.IOException

class UtilTest {

    @Test
    fun testSimplifyPath() {
        assertThat(Util.simplifyPath("/a/b/c/d")).isEqualTo("/a/b/c/d")
        assertThat(Util.simplifyPath("/a/../b/")).isEqualTo("/b")
        assertThat(Util.simplifyPath("/a/./b/c/../d")).isEqualTo("/a/b/d")
        assertThat(Util.simplifyPath("./a")).isEqualTo("/a")
        assertThat(Util.simplifyPath("")).isEqualTo("")
    }

    @Test
    @Throws(IOException::class)
    fun createParentDirTest() {
        val child = mock(File::class.java)
        val parent = mock(File::class.java)
        whenever(child.canonicalFile) doReturn child
        whenever(child.parentFile) doReturn parent
        whenever(parent.isDirectory) doReturn true
        Util.createParentDirs(child)
        verify(parent).mkdirs()
    }
}
